doc/debuginfo.txt

```diff
@@ -0,0 +1,187 @@
+Babeltrace Debug Info Analysis
+-----------------------------
+
+The babeltrace debug info analysis is a set of new features which
+allow mapping events from a trace to their location in source code.
+
+Prerequisites
+-------------
+
+In order to install a version of babeltrace with debug info support,
+the following software is required:
+
+    * libelf
+    * libdw
+
+The two libraries required are provided by the elfutils project
+(https://fedorahosted.org/elfutils/), and can be installed through the
+elfutils package on Ubuntu, Debian, RHEL, and others.
+
+Getting Babeltrace with Debug Info Support
+------------------------------------------
+
+The debug info feature branch of babeltrace can be found at
+https://github.com/abusque/babeltrace/tree/debuginfo
+
+The installation procedure is unchanged from other versions, and can
+be found in the README file.
+
+Compiling for Debug Info Analysis
+---------------------------------
+
+Traced programs on which debug info analysis is to be performed can be
+compiled in a few different ways, and still lead to useful results.
+
+Ideally, one should compile the program in debug mode, using simply
+the `-g` flag. This generates DWARF information, which is then used by
+babeltrace to map an event's source location to a file and line
+number, and the name of the surrounding function.
+
+If the executable is not compiled with `-g` enabled, and thus no DWARF
+information is available, babeltrace will use ELF symbols from the
+executable. Instead of providing source file, line number and function
+name, however, the analysis will give as a result the name of the
+nearest function symbol, plus an offset in bytes to the location in
+the executable from which the event originated.
+
+If the executable has neither ELF symbols nor DWARF information,
+babeltrace will be unable to map an event to its source location, and
+will simply display the instruction pointer (address), as in prior
+versions of babeltrace.
+
+Getting the Right Tracer
```

Comment:
I know this is pretty trivial but I just thought I'd justify regardless: I usually follow title case rules as specified by both the MLA and APA style guides, which usually require capitalizing all non-article words in headings.